data.base.api/src/main/java/ru/tinkoff/qa/neptune/data/base/api/queries/jdoql/JDOQLResultQueryParams.java

```diff
@@ -0,0 +1,185 @@
+     * </p>
+     * @param expression is a function that returns a {@link Expression}
+     * @return self-reference
+     */
+    public JDOQLResultQueryParams<T, Q> having(Function<Q, Expression<?>> expression) {
+        checkArgument(nonNull(expression), "Having expression should be defined as not a null value");
+        var having = expression.apply(persistableExpression);
+        checkNotNull(having);
+        this.havingExpression = having;
+        return this;
+    }
+
+    /**
+     * Sets result field to query. The sample below:
+     * <p>
+     * {@code byJDOQuery(QPerson.class)
+     *          .setResultField(qPerson -> qPerson.name)
+     * </p>
+     * @param expression is a function that returns a {@link Expression}
+     * @return self-reference
+     */
+    public JDOQLResultQueryParams<T, Q> setResultField(Function<Q, Expression<?>> expression) {
+        checkArgument(nonNull(expression), "Result field expression should be defined as not a null value");
+        var result = expression.apply(persistableExpression);
+        checkNotNull(result);
+        this.resultFields =  new Expression[] {result};
+        return this;
+    }
+
+
+    /**
+     * Adds result field to query. The sample below:
+     * <p>
+     * {@code byJDOQuery(QPerson.class)
+     *          .addResultField(qPerson -> qPerson.name)
+     * </p>
+     * @param expression is a function that returns a {@link Expression}
+     * @return self-reference
+     */
+    public JDOQLResultQueryParams<T, Q> addResultField(Function<Q, Expression<?>> expression) {
+        checkArgument(nonNull(expression), "Result field expression should be defined as not a null value");
+        var result = expression.apply(persistableExpression);
+        checkNotNull(result);
+        this.resultFields = ofNullable(this.resultFields)
+                .map(expressions -> add(expressions, result))
+                .orElseGet(() -> new Expression[]{result});
+        return this;
+    }
+
+    /**
+     * Defines whenever to select distinct results or not.
+     *
+     * @param distinct to select distinct results or not
+     * @return self-reference
+     */
+    public JDOQLResultQueryParams<T, Q> distinct(boolean distinct) {
+        this.distinct = distinct;
+        return this;
+    }
+
+    @Override
+    public <M extends JDOQLTypedQuery<T>> M  buildQuery(M m) {
+        checkArgument(nonNull(resultFields) && resultFields.length > 0,
+                "Should be defined at least one resul field");
+
+        ofNullable(where).ifPresent(m::filter);
+
+        if (nonNull(rangeStart) && nonNull(rangeEnd)) {
+            m.range(rangeStart, rangeEnd);
+        }
+
+        ofNullable(orderExpressions)
+                .ifPresent(orderExpressions1 -> {
+                    if (orderExpressions1.length > 0) {
+                        m.orderBy(orderExpressions1);
+                    }
+                });
+
+        ofNullable(groupByExpressions)
```

Comment:
```
ofNullable(groupByExpressions)
                .filter(gBy-> gBy.length > 0)
                .ifPresent(gBy-> m.groupBy(gBy));
```
I suggest like alluded to above.

Reply:
the result of the action is void, not an array of expressins